doc/design/service-discovery.md

```diff
@@ -0,0 +1,421 @@
+# Service Discovery
+
+## Summary and problem description
+
+Service discovery over IPC channel e.g. message queue or UNIX domain socket is not performant since larger data is
+transferred, which can lead to transmission of several frames. If lots of services are discovered at high-frequency
+e.g. at startup the IPC channel can become a bottleneck. Furthermore, a lesson learned from the iceoryx development
```

Comment:
Could you please remove the line starting with `Furthermore` or rephrase it that this is only necessary for safety environments where all the resources are created on startup anyway, e.g. by placing all the objects in a `StatusPort` only the respective app has access to. This would be sent to the application when it registers at RouDi. In a dynamic non safety scenario this is not possible and we should continue to use the IPC channel.

Reply:
Rephrased. I guess you're right when it comes to the objects in shared memory in the safety case. However, we still need to be able to send specific pointers to the `PoshRuntime`. IMHO we should have one "Sned me the pointer"-mechanism for both the safety and the dynamic case for simplicity. #611 discusses this in more detail. @elfenpiff and myself were pondering about a factory class that lives in shared memory to be able to utilise RAII.